src/net_processing.cpp

```diff
@@ -219,7 +219,16 @@ struct Peer {
     /** Work queue of items requested by this peer **/
     std::deque<CInv> m_getdata_requests GUARDED_BY(m_getdata_requests_mutex);
 
-    explicit Peer(NodeId id) : m_id(id) {}
+    /**
+     * Salt used to compute short IDs during transaction reconciliation.
+     * Salt is generated randomly per-connection to prevent linking of
+     * connections belonging to the same physical node.
+     * Also, salts should be different per-connection to prevent halting
+     * of relay of particular transactions due to collisions in short IDs.
+     */
+    const uint64_t m_local_recon_salt;
```

Comment:
6103b5cd

I think you can make this part of `ReconState`, at least a tiny memory saved for non-reconciliation peers.